### PR TITLE
feat!: mapping schema should not have quotes

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/model/definition.py [hunk 1]
```diff
@@ -533,7 +533,7 @@ def update_schema(
             if mapping_schema:
                 nested_set(
                     self.mapping_schema,
-                    tuple(str(part) for part in table.parts),
+                    tuple(part.sql(copy=False)[1: -1] for part in table.parts),
```
Comment (georgesittas) [inline sqlmesh/core/model/definition.py:536]: Is it guaranteed we'll always have quotes here?

**Review by georgesittas — APPROVED** [review]
Left a comment to consider. If there's no guarantee for the quotes we could wrap the slice in a lambda that checks for quotes to make the fix more robust (albeit a bit slower)

File: sqlmesh/core/model/definition.py (hunk 1 above)
Comment (tobymao) [reply sqlmesh/core/model/definition.py:536]: yes, because all tables are fqn quoted

**Comment by izeigerman** [comment]
I think this might require a migration

**Comment by tobymao** [comment]
@izeigerman that's why i upgraded sqlglot

**Pull request merged**